driver/fdo.go

```diff
@@ -11,18 +11,54 @@ import (
 	"fyne.io/fyne"
 )
 
-var iconExtensions = []string{".png", ".svg"}
+var (
+	iconExtensions = []string{".png", ".svg"}
+	iconTheme      string
+	iconSize       string
+)
+
+//fdoIconData is a structure that contains information about .desktop files
+type fdoIconData struct {
+	name     string // Application name
+	iconName string // Icon name
+	iconPath string // Icon path
+	exec     string // Command to execute application
+}
+
+//Returns the name associated with an fdo app
+func (idata *fdoIconData) Name() string {
+	if idata == nil {
```

Comment:
Checking for nil should not be needed here, it's reasonable to assume that our "self" type is not nil.